### PR TITLE
Add custom credo config file

### PR DESCRIPTION
**Pull request by romarickb**
#### Tasks

- [x] Use credo config file from https://github.com/mirego/credo_naming
- [x] Fix new credo rules
- [x] Update Readme

### CODE REVIEW

File: .credo.exs [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+           "Serializer",
+           ~r/^Helpers?$/i,
+           ~r/^Utils?$/i
+         ]},
+        {CredoNaming.Check.Consistency.ModuleFilename,
+         excluded_paths: ["config", "mix.exs", "priv", "test/support"],
+         acronyms: [
+           {"ElixirBoilerplateGraphQL", "elixir_boilerplate_graphql"},
+           {"GraphQL", "graphql"}
+         ]}
```
Comment (remi) [inline .credo.exs]: You’re going to need `credo_envvar` and `credo_naming` for those rules. And I think we can ditch `credo_envvar` here actually.

Comment (romarickb) [reply .credo.exs]: in `mix.exs` we have 

```elixir
defp deps do
  [
      # Linting
      {:credo, "~> 1.1", only: [:dev, :test], override: true},
      {:credo_envvar, "~> 0.1", only: [:dev, :test], runtime: false},
      {:credo_naming, "~> 0.4", only: [:dev, :test], runtime: false}
  ]
end
```

Comment (remi) [reply .credo.exs]: 🤦‍♂

File: .credo.exs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+        {Credo.Check.Warning.LazyLogging, false},
+        {Credo.Check.Warning.OperationOnSameValues},
+        {Credo.Check.Warning.BoolOperationOnSameValues},
+        {Credo.Check.Warning.UnusedEnumOperation},
+        {Credo.Check.Warning.UnusedKeywordOperation},
+        {Credo.Check.Warning.UnusedListOperation},
+        {Credo.Check.Warning.UnusedStringOperation},
+        {Credo.Check.Warning.UnusedTupleOperation},
+        {Credo.Check.Warning.OperationWithConstantResult},
+        {CredoEnvvar.Check.Warning.EnvironmentVariablesAtCompileTime},
```
Comment (remi) [inline .credo.exs]: I think we’ll need to remove this check.

File: .credo.exs [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        {CredoNaming.Check.Warning.AvoidSpecificTermsInModuleNames,
```
Comment (remi) [inline .credo.exs:59]: We could remove this check as well. I don’t think it’s going to be useful in this project.

File: .credo.exs [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+           "Persister",
+           "Serializer",
+           ~r/^Helpers?$/i,
+           ~r/^Utils?$/i
+         ]},
+        {CredoNaming.Check.Consistency.ModuleFilename,
+         excluded_paths: ["config", "mix.exs", "priv", "test/support"],
+         acronyms: [
+           {"ElixirBoilerplateGraphQL", "elixir_boilerplate_graphql"},
+           {"GraphQL", "graphql"}
```
Comment (remi) [inline .credo.exs]: We probably should update these values too.

**Pull request merged**